File: python/2215.find-the-difference-of-two-arrays.py
#
# @lc app=leetcode id=2215 lang=python3
#
# [2215] Find the Difference of Two Arrays
#

# @lc code=start
from typing import List


class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        s1 = set(nums1)
        s2 = set(nums2)

        l1 = []
        l2 = []
        for val in nums1:
            if val not in s2 and val not in l1:
                l1.append(val)

        for val in nums2:
            if val not in s1 and val not in l2:
                l2.append(val)

        return [l1, l2]


obj = Solution()
print(obj.findDifference([1, 2, 3], [2, 4, 6]))
print(obj.findDifference([1, 2, 3, 3], [1, 1, 2, 2]))

# @lc code=end

File: 2215.find-the-difference-of-two-arrays.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/*
 * @lc app=leetcode id=2215 lang=java
 *
 * [2215] Find the Difference of Two Arrays
 */

// @lc code=start
class Solution {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Map<Integer, Integer> map1 = new HashMap<>();
        Map<Integer, Integer> map2 = new HashMap<>();

        for (int val : nums1) {
            map1.put(val, map1.getOrDefault(val, 0) + 1);
        }

        for (int val : nums2) {
            map2.put(val, map2.getOrDefault(val, 0) + 1);
        }

        List<Integer> list1 = new ArrayList<>();
        for (int val : nums1) {
            if (!map2.containsKey(val) && !list1.contains(val))
                list1.add(val);
        }

        List<Integer> list2 = new ArrayList<>();
        for (int val : nums2) {
            if (!map1.containsKey(val) && !list2.contains(val))
                list2.add(val);
        }

        return List.of(list1, list2);
    }
}
// @lc code=end
